bin/process_dbcan_result.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import os
+import sys
+
+logging.basicConfig(level=logging.INFO)
+
+
+def main(input_folder, outfile, dbcan_version):
+    if not check_folder_completeness(input_folder):
+        sys.exit("Missing dbCAN outputs. Exiting.")
+    substrates = load_substrates(input_folder)
+    cgc_locations = load_cgcs(input_folder)
+    print_gff(input_folder, outfile, dbcan_version, substrates, cgc_locations)
+
+
+def load_cgcs(input_folder):
+    cgc_locations = dict()
+    with open(os.path.join(input_folder, "cgc_standard.out")) as file_in:
+        for line in file_in:
+            if not line.startswith("CGC#"):
+                cgc, _, contig, _, start, end, _, _ = line.strip().split("\t")
+                if cgc in cgc_locations:
+                    if cgc_locations[cgc]["start"] > int(start):
+                        cgc_locations[cgc]["start"] = int(start)
+                    if cgc_locations[cgc]["end"] < int(end):
+                        cgc_locations[cgc]["end"] = int(end)
+                else:
+                    cgc_locations[cgc] = {"start": int(start),
+                                          "end": int(end),
+                                          "contig": contig}
+    return cgc_locations
+
+
+def print_gff(input_folder, outfile, dbcan_version, substrates, cgc_locations):
```

Comment:
I think this method can be more pythonic:
`import csv

def print_gff(input_folder, outfile, dbcan_version, substrates, cgc_locations):

    with open(outfile, "w") as f_out:
        writer = csv.writer(f_out, delimiter="\t")
        
        writer.writerow(["##gff-version 3"])
        
        printed_cgcs = []
        
        with open(os.path.join(input_folder, "cgc_standard.out")) as f_in:
            reader = csv.reader(f_in, delimiter="\t")
            for line in reader:
                if line[0].startswith("CGC#"):
                    continue
                    
                cgc, gene_type, contig, prot_id, start, end, strand, protein_fam = line
                
                if cgc not in printed_cgcs:
                    substrate = substrates.get(cgc, "substrate_dbcan_pul=N/A;substrate_ecami=N/A")
                    
                    writer.writerow([contig, f"dbCAN:{dbcan_version}", "predicted PUL", 
                                    cgc_locations[cgc]['start'], cgc_locations[cgc]['end'], 
                                    ".", ".", ".", f"ID={cgc};{substrate}"])
                    printed_cgcs.append(cgc)
                    
                writer.writerow([contig, f"dbCAN:{dbcan_version}", gene_type, start, end, 
                                 ".", strand, ".", f"ID={prot_id};Parent={cgc},protein_family={protein_fam}"])`

**this is not tested**